nodejs/eks/cluster.ts

```diff
@@ -80,6 +80,29 @@ export interface CoreData {
     vpcCni?: VpcCni;
 }
 
+function translateAWSProviderToIAMAuth(clusterName: string, providerProfile?: string, providerAssumeRole?: { roleArn?: string }): pulumi.Output<any> {
+    const command = "aws-iam-authenticator";
+    const args = ["token", "-i", clusterName];
+    return pulumi.all([providerProfile, providerAssumeRole]).apply(([profile, assumeRole]) => {
+        const result: {[k: string]: any} = { args, command, apiVersion: "client.authentication.k8s.io/v1alpha1" };
+
+        if (typeof assumeRole === "string") {
```

Comment:
Can this ever be true?  The TypeScript type doesn't think so, and the typing of the `assumeRole` in the provider doesn't appear to have any case where this would be a string.

Reply:
For some reason, it was a serialized json. I know that in the configuration you should specify the `assumeRole` as json